packages/office-ui-fabric-react/src/utilities/keytips/KeytipManager.ts

```diff
@@ -0,0 +1,430 @@
+    this.keytipTree.addNode(keytipProps, uniqueID, true);
+    return uniqueID;
+  }
+
+  /**
+   * Update a keytip's props
+   *
+   * @param keytipProps - Keytip to update
+   * @param uniqueID - Unique ID of this keytip
+   */
+  public updateKeytip(keytipProps: IKeytipProps, uniqueID: string): void {
+    // Update keytip in this.keytips
+    const uniqueKeytip = this._constructUniqueKeytip(keytipProps, uniqueID);
+    const keytipIndex = this._findKeytipIndex(uniqueKeytip);
+    if (keytipIndex >= 0) {
+      this._updateKeytip(uniqueKeytip, keytipIndex);
+    }
+  }
+
+  /**
+   * Unregisters a keytip
+   *
+   * @param keytipToRemove - IKeytipProps of the keytip to remove
+   */
+  public unregisterKeytip(keytipToRemove: IKeytipProps, uniqueID: string): void {
+    // Add the unique ID to the keytip
+    const uniqueKeytipToRemove = this._constructUniqueKeytip(keytipToRemove, uniqueID);
+
+    // Remove keytipToRemove from this.keytips
+    this.keytips = this.keytips.filter((uniqueKtp: IUniqueKeytip) => {
+      return !(fullKeySequencesAreEqual(uniqueKtp.keytip.keySequences, keytipToRemove.keySequences) && uniqueKtp.uniqueID === uniqueID);
+    });
+
+    // Remove keytip from the delayed queue
+    this._removeKeytipFromQueue(convertSequencesToKeytipID(keytipToRemove.keySequences));
+
+    // Remove the node from the Tree
+    this.keytipTree.removeNode(keytipToRemove, uniqueID);
+    this._layer && this._layer.setKeytips(this.getKeytips());
+  }
+
+  /**
+   * Unegister a persisted keytip
+   * This means just removing it from the KeytipTree
+   *
+   * @param keytipToRemove - Persisted keytip to remove
+   * @param uniqueID - Unique ID of this persisted keytip
+   */
+  public unregisterPersistedKeytip(keytipToRemove: IKeytipProps, uniqueID: string): void {
+    // Add the unique ID to the keytip
+    this.keytipTree.removeNode(keytipToRemove, uniqueID);
+  }
+
+  /**
+   * Method that makes visible keytips currently in the DOM given a list of IDs.
+   *
+   * @param ids - list of Keytip IDs to show
+   */
+  public showKeytips(ids: string[]): void {
+    // Set visible property in this.keytips
+    for (const keytip of this.getKeytips()) {
+      const keytipId = convertSequencesToKeytipID(keytip.keySequences);
+      if (ids.indexOf(keytipId) >= 0) {
+        keytip.visible = true;
+      } else if (keytip.overflowSetSequence && ids.indexOf(
+        convertSequencesToKeytipID(
+          mergeOverflowKeySequences(keytip.keySequences, keytip.overflowSetSequence))) >= 0) {
+        // Check if the ID with the overflow is the keytip we're looking for
+        keytip.visible = true;
+      } else {
+        keytip.visible = false;
+      }
+    }
+
+    // Set in layer
+    this._layer && this._layer.setKeytips(this.getKeytips());
+  }
+
+  /**
+   * Exit keytip mode
+   */
+  public exitKeytipMode(): void {
+    this.keytipTree.currentKeytip = undefined;
+    // Hide all keytips
+    this.showKeytips([]);
```

Comment:
>this.showKeytips([]); [](start = 4, length = 21)

i wonder if it is nicer to have a separate function to do this, or just in place here have a foreach to set all keytips to not visible and update the layer